testhelper/docker/resource/etcd/etcd.go

```diff
@@ -0,0 +1,67 @@
+package etcd
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/ory/dockertest/v3"
+	etcd "go.etcd.io/etcd/client/v3"
+	"google.golang.org/grpc"
+
+	"github.com/rudderlabs/rudder-go-kit/testhelper/docker/resource"
+)
+
+type Resource struct {
+	Client *etcd.Client
+	Hosts  []string
+	Port   int
+}
+
+func Setup(pool *dockertest.Pool, cln resource.Cleaner) (*Resource, error) {
+	etcdImage := "bitnami/etcd"
+	container, err := pool.Run(etcdImage, "3.5", []string{
```

Comment:
Since we already have options defined, we can probably leverage it here?
```suggestion
func Setup(pool *dockertest.Pool, cln resource.Cleaner, opts ...func(*Config)) (*Resource, error) {
	c := &Config{
		Tag: "bitnami/etcd",
	}
	for _, opt := range opts {
		opt(c)
	}
	container, err := pool.Run(c.Tag, "3.5", []string{
```